test/cppsuite/tests/background_compact.cpp

```diff
@@ -0,0 +1,362 @@
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "src/common/constants.h"
+#include "src/common/logger.h"
+#include "src/common/random_generator.h"
+#include "src/main/test.h"
+#include "src/main/validator.h"
+
+namespace test_harness {
+
+/* This test produces a workload that encourages the background compaction server to do work by:
+ *      1. Performing random truncations that remove 20% of keys on randomly selected tables.
+ *      2. Providing a "maintenance window" which allows compact to continue running while all other
+ *         operations are paused. The period of the maintenance window is set by the custom
+ *         operations op_rate.
+ *      3. Performing inserts to ensure the files continue to grow.
+ */
+class background_compact : public test {
+    bool maintenance_window = false;
+
+public:
+    background_compact(const test_args &args) : test(args)
+    {
+        init_operation_tracker();
+    }
+
+    /* Custom operation to simulate toggling maintenance windows in a workload. */
+    void
+    custom_operation(thread_worker *tw) override final
+    {
+        std::string log_prefix =
+          type_string(tw->type) + " thread {" + std::to_string(tw->id) + "}: ";
+        logger::log_msg(
+          LOG_INFO, type_string(tw->type) + " thread {" + std::to_string(tw->id) + "} commencing.");
+
+        while (tw->running()) {
+            maintenance_window = !maintenance_window;
+            std::string state = maintenance_window ? "On" : "Off";
+            logger::log_msg(LOG_TRACE, log_prefix + " toggle maintenance window " + state);
+
+            tw->sleep();
+        }
+    }
+
+    void
+    remove_operation(thread_worker *tw) override final
+    {
+        std::string log_prefix =
+          type_string(tw->type) + " thread {" + std::to_string(tw->id) + "}: ";
```

Comment:
Is this log_prefix standard in the CPP suite?

Reply:
No